hack/godep-save.sh

```diff
@@ -21,9 +21,27 @@ set -o pipefail
 FRAKTI_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${FRAKTI_ROOT}/hack/lib/init.sh"
 GODEP="${GODEP:-godep}"
+PRE_RESTORE_FLAG="--pre-restore"
 
+# Some special pkg dependencies we may need restore them
+# before godep-save when try to update
+PRE_RESTORE_PKG=(
+  "k8s.io/client-go"
+)
 
-cd ${FRAKTI_ROOT}
+if [ $# -gt 0 ]; then
+  if [ "$1" = "$PRE_RESTORE_FLAG" ]; then
+    for pkg in ${PRE_RESTORE_PKG[@]}
+    do
+      cd ${GOPATH}/src/$pkg
```

Comment:
> what version should ${GOPATH}/src/$pkg be?

depends on user, keep original version or update to a new version. Here just ensure its dependencies are restored properly(just like `client-go` relay on `apimachinery`)

> cd ${GOPATH}/src/$pkg may fail because it doesn't exist

I think fail and exit is expected if this pkg doesn't exist. And if this pkg is not required anymore, should remove it from `PRE_RESTORE_PKG` too